### ISSUE TITLE
IPV6 announcement fails on Linux

### ISSUE THREAD

**Issue by rolznz** [videolabs/libmicrodns#23]
To reproduce, change the announce example to init libmicrodns with MDNS_ADDR_IPV6 instead of MDNS_ADDR_IPV4.

Binding the socket will fail with "invalid argument"

```
if (bind(ctx->conns[i].sock, (const struct sockaddr *) &ctx->addr, ss_len(&ctx->addr)) < 0)
                    return mdns_destroy(ctx), (MDNS_NETERR);
```

The socket is created for the correct address family (AF_INET6) and ss_len(&ctx->addr) matches sizeof(struct sockaddr_in6)

I'm trying to figure out what's wrong but so far not having any luck. If anyone has any ideas, let me know.

**Comment by 3XX0** [comment]
Code changed quite a bit since I last looked at it and I don't really understand all the interface selection logic, so I will leave the patch up to @chouquette :)

As far as IPv6 goes:

- `bind` needs an interface index for IPv6 mcast addresses (field `sin6_scope_id` needs to be set)
- `IPV6_MULTICAST_LOOP` should replace `IP_MULTICAST_LOOP` and it expects a 4 byte integer
- `IPV6_MULTICAST_IF` expects an interface index as an integer

**Comment by chouquette** [comment]
This should be fixed by #35
